cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/management-credential-step/ManagementCredentials.tsx

```diff
@@ -153,6 +170,57 @@ function ManagementCredentials(props: Partial<StepProps>) {
         }
     };
 
+    useEffect(() => {
+        retrieveOsImages(awsState[STORE_SECTION_FORM].REGION);
+    }, [awsState[STORE_SECTION_FORM].REGION]);
+
+    function retrieveOsImages(region: string | undefined) {
+        try {
+            setOsImages([]);
+            AwsService.getAwsosImages(region).then((data) => {
+                setOsImages(data);
+                setErrorMessage((errorMessage) => {
+                    const copy = { ...errorMessage };
+                    delete copy['OS_IMAGE'];
+                    return copy;
+                });
+            });
+        } catch (e) {
+            setErrorMessage({
```

Comment:
I'm not sure `errorMessage` will have a value here. Is this supposed to be a closure like line 182 (with `(errorMessage) =>`)?

Reply:
Using closure can work. I think `errorMessage` here also has a value. I will test it. Thanks

Reply:
For the concerns of 'if errorMessage has value or not', I have tested it manually.
If you are asking if all fields in errorMessage will be copied to new object, the answer is yes.
We can also use JSON.stringify() to copy nested object.
But it depends on the structure of errorMessage. If we want to store error info like errorName: {code:"", info:""}, we can use JSON.stringify() instead